Hudl.Mjolnir/Bulkhead/SemaphoreBulkhead.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Hudl.Mjolnir.Key;
+using System;
+using System.Threading;
+
+namespace Hudl.Mjolnir.Bulkhead
+{
+    internal interface IBulkheadSemaphore
+    {
+        void Release();
+        bool TryEnter();
+        int Available { get; }
```

Comment:
Do we want to rename this to `CountAvailable` or something indicative of it being a count?